lib/sqlite.js

```diff
@@ -3,3 +3,8 @@ const { emitExperimentalWarning } = require('internal/util');
 
 emitExperimentalWarning('SQLite');
 module.exports = internalBinding('sqlite');
+
+const statementIterate = module.exports.StatementSync.prototype.iterate;
+module.exports.StatementSync.prototype.iterate = function iterate() {
+  return Iterator.from(statementIterate.apply(this, arguments));
```

Comment:
use `SafeArrayIterator`

Reply:
What is `SafeArrayIterator`, and where to use it / How to access it ?
Is it a `v8` Class, a node cpp builtin Class ?

Reply:
https://github.com/nodejs/node/blob/3cbeed88d9de9231f420ba53ad0e4d4a7af9c637/doc/contributing/primordials.md?plain=1#L107

Reply:
Thanks, I'll take a look

Reply:
`StatementSync::iterate` return an iterator, not an iterable. So if I wrap it with `new SafeArrayIterator(statementIterate.apply(this, arguments))`

1. the iterator is empty
  ```
  > iterator.next()
  { value: undefined, done: true } // it should return `{ value: {a: 1}, done: false }`
  ```
2. and `iterator` is not instance of `Iterator`, so no integration with `IteratorHelpers`
  ```
  > iterator instanceof Iterator
  false
  > iterator.toArray
  undefined
  ```

So even if I implement iterable protocol in `StatementSync::iterate`, it should fix the iterator (not empty). But I'll need to keep `Iterator.from` decoration to keep IteratorHelper in the iterator.

Is it OK for you if final code is:

```js
module.exports.StatementSync.prototype.iterate = function iterate() {
  return Iterator.from(
    new SafeArrayIterator(statementIterate.apply(this, arguments))
  );
};
```

Reply:
I implemented iterable protocol in cpp:

```cpp
void IteratorFunc(const v8::FunctionCallbackInfo<v8::Value>& info) {
  info.GetReturnValue().Set(info.Holder());
}
// ...
  v8::Local<v8::FunctionTemplate> iteratorFuncTemplate =
      v8::FunctionTemplate::New(isolate, IteratorFunc);
  iterableIteratorTemplate->Set(v8::Symbol::GetIterator(isolate),
                                iteratorFuncTemplate);
```

I tried it without wrapping it, and worked as expected. `iterator.next() // {value: {a: 1}, done: false}`

I wrap it with `new SafeArrayIterator(statementIterate.apply(this, arguments))` and the iterator is still empty.
`iterator.next() // {value: undefined, done: true}`

I guess `SafeArrayIterator` is not adapted for a raw IterableIterator `{next() {}, [Symbol.iterator]() {return this}}`.
So I'll keep working `return Iterator.from(statementIterate.apply(this, arguments))`.

Reply:
I found how to do it in cpp, so no need to do hacky js finally